src/main/java/core/strategy/TransactionProcessor.java

```diff
@@ -0,0 +1,8 @@
+package core.strategy;
+
+import core.db.StorageDao;
+import core.model.FruitTransaction;
+
+public interface TransactionProcessor {
+    void process(FruitTransaction transaction, StorageDao storageDao);
```

Comment:
```suggestion
    void process(FruitTransaction transaction);
```
Мы не передаем ДАО аргументом метода